core/src/main/scala/kanaloa/reactive/dispatcher/queue/QueueProcessor.scala

```diff
@@ -65,7 +65,13 @@ class QueueProcessor(
 
     case Terminated(worker) if workerPool.contains(worker) ⇒
       removeWorker(worker)
-      healthCheck()
+      if (workerPool.length < settings.minPoolSize) {
+        log.error("Worker death caused worker pool size drop below minimum.")
+        if (workerPool.isEmpty && settings.shutdownOnAllWorkerDeath) {
+          log.error("Dispatcher is shutdown because there are no workers left.")
+          context stop self //this will also shutdown the whole dispatcher
```

Comment:
Do we need something more graceful?  IE: should this just enter the shutdown state, that way it behaves like all other shutdown behavior?   I'm only thinking of a general case where there is Work inside the Queue which will need to be rejected/failed